File: anm_backend/assistant/leticia/dialogue/mode_resolver.py
"""Resolve dialogue mode from a classified intent."""

from __future__ import annotations

from anm_backend.assistant.leticia.types import LeticiaDialogueMode, LeticiaIntent


def resolve_leticia_dialogue_mode(intent: LeticiaIntent) -> LeticiaDialogueMode:
    if intent.name in {"greeting", "checkin", "gratitude", "farewell", "confirmation", "negation"}:
        return "social"
    if intent.name == "help_request":
        return "assist"
    if intent.name == "command":
        return "command"
    if intent.name == "question":
        return "direct_answer"
    if intent.name == "ambiguous":
        return "clarify"
    return "clarify" if intent.is_micro_turn else "contextual"

File: core/leticia/dialogue/mode-resolver.ts
import type { LeticiaDialogueMode, LeticiaIntent } from "../types";

export function resolveLeticiaDialogueMode(intent: LeticiaIntent): LeticiaDialogueMode {
  switch (intent.name) {
    case "greeting":
    case "gratitude":
    case "farewell":
    case "confirmation":
    case "negation":
      return "social";
    case "help_request":
      return "assist";
    case "command":
      return "command";
    case "question":
      return "direct_answer";
    case "ambiguous":
      return "clarify";
    default:
      return intent.isMicroTurn ? "clarify" : "contextual";
  }
}

